src/LMTR_alg.jl

```diff
@@ -105,6 +105,7 @@ function LMTR(
   Jk = jac_op_residual(nls, xk)
   ∇fk = Jk' * Fk
   JdFk = similar(Fk)   # temporary storage
+  Jt_Fk = similar(∇fk)   # temporary storage
```

Comment:
Is storing faster than taking the transpose?

Reply:
I'm just trying to avoid allocating an array for the result each time we need a transposed-Jacobian-vector product. Sorry, not sure I understand the question.

Reply:
`Jk` is just an operator, so if we write `Jk' * Fk`, a new vector will be allocated each time.